gtk/src/callbacks.c

```diff
@@ -577,6 +579,70 @@ check_name_template(signal_user_data_t *ud, const char *str)
     return FALSE;
 }
 
+int match(const char *string, const char *pattern)
+{
+    int status;
+    regex_t re;
+    if (regcomp(&re, pattern, REG_EXTENDED|REG_NOSUB) != 0)
+    {
+        return(0);
+    }
+    status = regexec(&re, string, (size_t) 0, NULL, 0);
+    regfree(&re);
+    if (status != 0) 
+    {
+        return(0);
+    }
+    return(1);
+}
+
+typedef struct {
+    const char *pattern;
+    const char *format;
+} datemap;
+
+static int
+parse_datestring(const char *src, struct tm *tm)
+{       
+    datemap ymdThmsZ = {"[0-9]{4}-[0-1]?[0-9]-[0-3]?[0-9]T[0-9]{2}:[0-9]{2}:[0-9]{2}Z", "%Y-%m-%dT%H:%M:%SZ"};  
+
+    datemap maps[1] = { ymdThmsZ };
+
+    for (int i = 0; i < sizeof(maps); i++)
+    {
+        if (match(src, maps[i].pattern))
+        {
+            strptime(src, maps[i].format, tm);
+            return(1);
+        }
+    }
+    return(0);
+}
+
+static char*
+get_creation_date(const char *pattern, const char *metaValue, const char *file, char **str)
+{       
+    char date[11];       
```

Comment:
date may not get initialized in this function.  I suggest setting to empty string or something to indicate no date was available.